indent/rust.vim

```diff
@@ -272,6 +272,31 @@ function GetRustIndent(lnum)
         endif
     endif
 
+    " cindent by messes up indentation of parentheses. It does so regardless of
+    " the cindent options.
+    "
+    " Consider this input example, where | is the cursor:
+    "
+    "     foo(|)
+    "
+    " When pressing Enter, the indentation will be this:
+    "
+    "     foo(
+    "         |)
+    "
+    " But that's not what we want, we want it to be this instead:
+    "
+    "     foo(
+    "     |)
+    if line =~# '\V\^\s\*)'
+        let l:paren_start = searchpair('(', '', ')', 'nbW',
+            \ 's:is_string_comment(line("."), col("."))')
+
+        if l:paren_start != 0 && l:paren_start < a:lnum
+            return indent(l:paren_start)
+        endif
+    endif
```

Comment:
Frankly, I'm not sure if this it _the_ best way of going about fixing this, but it seemed to be the least problematic approach. I spent some time trying to figure out if there's another way (e.g. setting some option), but it seems that enabling `cindent` results in this behaviour, without there being an obvious way of disabling the indentation of `)`.